src/main/java/org/snomed/snowstorm/core/data/services/BranchReviewService.java

```diff
@@ -228,6 +231,81 @@ public void applyMergeReview(MergeReview mergeReview) throws ServiceException {
 		branchMergeService.mergeBranchSync(mergeReview.getSourcePath(), mergeReview.getTargetPath(), concepts);
 	}
 
+	private Concept upgradeComponents(Concept sourceVersion, Concept targetVersion) {
+		// Copy from latest component
+		Concept winningConcept = sourceVersion.isReleasedMoreRecentlyThan(targetVersion) ? sourceVersion : targetVersion;
+
+		Concept upgradedConcept = new Concept();
+		upgradedConcept.setConceptId(winningConcept.getConceptId());
+		upgradedConcept.setEffectiveTimeI(winningConcept.getEffectiveTimeI());
+		upgradedConcept.setActive(winningConcept.isActive());
+		upgradedConcept.setReleased(winningConcept.isReleased());
+		upgradedConcept.setReleaseHash(winningConcept.getReleaseHash());
+		upgradedConcept.setReleasedEffectiveTime(winningConcept.getReleasedEffectiveTime());
+		upgradedConcept.setModuleId(winningConcept.getModuleId());
+		upgradedConcept.setDefinitionStatusId(winningConcept.getDefinitionStatusId());
+		upgradedConcept.setInactivationIndicator(winningConcept.getInactivationIndicator());
+		upgradedConcept.setAssociationTargets(winningConcept.getAssociationTargets());
+		upgradedConcept.setDescriptions(mergePublished(sourceVersion.getDescriptions(), targetVersion.getDescriptions()));
+		upgradedConcept.setRelationships(mergePublished(sourceVersion.getRelationships(), targetVersion.getRelationships()));
+		upgradedConcept.setClassAxioms(mergePublishedAxioms(sourceVersion.getClassAxioms(), targetVersion.getClassAxioms()));
+		upgradedConcept.setGciAxioms(mergePublishedAxioms(sourceVersion.getGciAxioms(), sourceVersion.getGciAxioms()));
+
+		return upgradedConcept;
+	}
+
+	private <T extends SnomedComponent> Set<T> mergePublished(Set<T> sourceComponents, Set<T> targetComponents) {
+		Map<String, T> mergedComponents = new HashMap<>();
+
+		for (T sourceComponent : sourceComponents) {
+			boolean publishedAndUnchanged = sourceComponent.getReleasedEffectiveTime() != null && sourceComponent.getEffectiveTimeI() != null;
+			if (publishedAndUnchanged) {
```

Comment:
sourceComponent.getEffectiveTimeI() != null is sufficient

Reply:
After further checking the logic here is not quite right. We should only replace the target with the source version when both versions have effective time and the source version is newer than the target. I have drafted the logic and will send it to you.